### PR TITLE
Bug 1215219 - Integration tests for Late Customization

### PR DESCRIPTION
**Pull request by sfoster**
Test the "Free Apps" panel isn't shown when we got empty response. 
Test a late customization response results in the apps showing up in the Free Apps panel, and getting installed for reals

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=945b62ad998b7fc70a1f19698fb8f728525e8856

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d08782abc6de22daf88b1a1a677aaf3204dadfad

File: apps/ftu/test/marionette/late_customization_test.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+      test('hide panel with no apps', () => {
+        // click through to finish, ensure free apps panel is not displayed
+        var finishScreen = client.findElement(Ftu.Selectors.finishScreen);
+        var seenLCPanel = false;
+        var hash;
+        while (!seenLCPanel && !finishScreen.displayed()) {
+          hash = ftu.getLocationHash();
+          if (hash.indexOf(lateCustom.hash) === 0) {
+            seenLCPanel = true;
```
Comment (arcturus) [inline apps/ftu/test/marionette/late_customization_test.js]: tiny nit:
```javascript
seenLCPanel = hash.indexOf(lateCustom.hash) === 0;
```

File: apps/ftu/test/marionette/late_customization_test.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+      test('FTU operates normally without valid server response', () => {
+        // click through to finish, ensure free apps panel is not displayed
+        var finishScreen = client.findElement(Ftu.Selectors.finishScreen);
+        var seenLCPanel = false;
+        var hash;
+        while (!seenLCPanel && !finishScreen.displayed()) {
+          hash = ftu.getLocationHash();
+          if (hash.indexOf(lateCustom.hash) === 0) {
+            seenLCPanel = true;
```
Comment (arcturus) [inline apps/ftu/test/marionette/late_customization_test.js]: same than above

File: apps/ftu/test/marionette/lib/server-parent.js [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+/* global module, __dirname */
+var fork = require('child_process').fork;
+/**
+ * Issue a POST request via marionette.
+ */
+function post(client, url, json) {
+  // must run in chrome so we can do cross domain xhr
+  client = client.scope({ context: 'chrome' });
+  return client.executeAsyncScript(function(url, json) {
+    var xhr = new XMLHttpRequest();
```
Comment (arcturus) [inline apps/ftu/test/marionette/lib/server-parent.js]: nit: potentially we could use fetch api

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=179b9b477041980459aac055bb10dcffed4dbdba

**Pull request merged**